### PR TITLE
Enable prow on csi-proxy repo

### PR DESCRIPTION
**Pull request by msau42**
This will add a required job that builds and runs unit tests

### CODE REVIEW

**Comment by msau42** [comment]
/hold
waiting on csi-proxy release-tools setup

**Comment by chases2** [comment]
It looks like you need something like 

```
annotations:
  testgrid_dashboards: <dashboard>
```

See Testgrid's [config doc](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md) for more information

File: config/jobs/kubernetes-csi/csi-proxy/csi-proxy-config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# generated by gen-jobs.sh, do not edit manually
+
+presubmits:
+  kubernetes-csi/csi-proxy:
+  - name: pull-kubernetes-csi-csi-proxy
+    always_run: true
+    decorate: true
+    skip_report: false
+    skip_branches: []
+    labels:
```
Comment (chases2) [inline config/jobs/kubernetes-csi/csi-proxy/csi-proxy-config.yaml]: It looks like in this job config you need something like 

```
annotations:
  testgrid-dashboards: <dashboard>
```

See Testgrid's [config doc](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md) for more information

**Comment by msau42** [comment]
```
--- FAIL: TestKubernetesProwInstanceJobsMustHaveMatchingTestgridEntries (5.17s)
    config_test.go:463: Testgrid group pull-kubernetes-csi-csi-driver-flex does not have a matching jenkins or prow job
    config_test.go:463: Testgrid group pull-kubernetes-csi-csi-lib-utils does not have a matching jenkins or prow job
    config_test.go:463: Testgrid group pull-kubernetes-csi-csi-test does not have a matching jenkins or prow job
    config_test.go:463: Testgrid group pull-kubernetes-csi-csi-release-tools does not have a matching jenkins or prow job
```

Hi @chases2, I have a script that generates the configs across multiple repos.  So when I added the testgrid annotations to the new repo "csi-proxy", it also generated testgrid annotations on existing repos.  The error message is a little confusing to me because there are prow jobs defined for those.  Could you help clarify what the issue is?

**Comment by chases2** [comment]
Needs rebase (sorry), but this looks good to me
/assign

**Comment by chases2** [comment]
Note: testgrid/cmd/configurator/config_test.go went to [config/tests/testgrids/config_test.go](https://github.com/kubernetes/test-infra/blob/master/config/tests/testgrids/config_test.go), but did not substantially change

**Comment by msau42** [comment]
rebased

**Pull request merged**